default_settings.lua

```diff
@@ -70,6 +70,7 @@ for name, value in pairs(settings) do
 	elseif setting_type == "number" then
 		pipeworks[name] = tonumber(minetest.settings:get(prefix..name) or value)
 	else
-		pipeworks[name] = value
+		local got_setting = minetest.settings:get(prefix..name)
+		pipeworks[name] = got_setting ~= "" and got_setting or value
```

Comment:
This will produce nil settings, because `get()` returns nil if there is no value set, not an empty string. It should be like this:
```suggestion
		pipeworks[name] = minetest.settings:get(prefix..name) or value
```

Reply:
Didn't catch this one but this was added for lua tube print configuration, however as it is now done similar to mesecons allowing run time configuration this change isn't actually needed at all and could be removed.

I think this main configuration thing could see rework at some point so that at least new settings would go into separate configuration table instead of straight under `pipeworks`. This would then also allow mixing run time and load time configs under same table and same API (similar how `pipeworks.enable_*` etc. currently works).

If doing that then it should anyway be in its own PR and for this one could also just revert this change / this file to its original state:
```suggestion
		pipeworks[name] = value
```

I'd be fine also with changes, suggestion is mainly because after applying other suggestions this change does not actually do anything anymore. (basically reset to master for this file would work as well).

Reply:
My approval stays no matter how this gets handled / if this gets updated or not, don't really care if few lines of unused code is updated to more correct version of unused code 😄

Reply:
I have a mod for settings reloading: https://github.com/C-C-Minetest-Server/settings_loader

It is still a work in progress, but is already performing well on my few mods loading basic values like strings and integers.